File: Estágio AutoMind/Usuario/BuscarUsuario.cs
namespace EstagioAutomind.Usuario;

public class BuscarUsuario {
	/* Método para buscar usuário na lista a partir do nome */
	public static void buscar(List<CadastroUsuario> usuario) {
		try {
			if (usuario.Count == 0)
				throw new Exception("A lista está vazia, a seção cadastrado será acessada.");
			
			Console.Write("\nDigite o nome do usuário que deseja buscar: ");
			string nome = Console.ReadLine().Trim().ToLower();
			/* ToLower() -> Transforma todos os caracteres da expressão (argumento) em letra minúscula
			   Trim() -> Remove caracteres especificados do lado que houver (esquerdo ou direito) */
			
			/* Operação lambda para busca na lista */
			var usuariosEncontrados =
				usuario.Where(u => u.Nome.Trim().ToLower().Contains(nome)).ToList();
			/* Contains(nome) -> Busca o nome mesmo se houver somente partes dele */
			
			bool encontrado = false; /* Booleano para visualizar situação da busca */
			
			foreach (var u in usuariosEncontrados) {
				if (usuariosEncontrados.Any()) {
					Console.WriteLine();
					Console.WriteLine($"Nome: {u.Nome}");
					Console.WriteLine($"Email: {u.Email}");
					Console.WriteLine($"Idade: {u.Idade}");
					encontrado = true;
				} 
			}
			
			if (!encontrado) { Console.WriteLine("Usuário inexistente!"); }
		} catch (Exception e) {
			/* Capturamos o erro caso a lista esteja vazia */
			Console.WriteLine($"Erro: {e.Message}");
			
			/* Em caso de lista vazia, seção de cadastro é chamada automaticamente */
			CadastroUsuario.Cadastro(usuario);
		}
	}
}

File: Estágio AutoMind [Java]/src/Usuario/BuscarUsuario.java
package Usuario;

import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class BuscarUsuario {
    public static void buscar(List<CadastroUsuario> usuarios, Scanner sc) {
        try {
            if (usuarios.isEmpty())
                throw new Exception("Lista vazia, seção cadastro será acessada automaticamente!");

            System.out.print("Digite o nome completo ou parte do que deseja buscar: ");
            String nome = sc.nextLine().trim().toLowerCase();

            List<CadastroUsuario> usuarioEncontrado =
                  usuarios.stream().filter(u -> u.getNome().toLowerCase().contains(nome)).toList();

            boolean encontrado = false;
            for (CadastroUsuario u : usuarioEncontrado) {
                System.out.println();
                System.out.println("Nome: " + u.nome);
                System.out.println("Email: " + u.email);
                System.out.println("Idade: " + u.idade);
                encontrado = true;
            }

            if (!encontrado) { System.out.println("Usuário não existente!"); }

        } catch (Exception e) {
            System.out.println(e.getMessage());
            CadastroUsuario.cadastro(usuarios, sc);
        }
    }
}
